workers/task-driver/src/tasks/settle_match_external.rs

```diff
@@ -348,14 +348,36 @@ impl SettleMatchExternalTask {
 
     /// Await the result of the atomic match settlement to be submitted on-chain
     ///
-    /// Returns `true` if the settlement landed on-chain, `false` otherwise
+    /// Returns `true` if the settlement succeeded on-chain, `false` otherwise
     async fn await_settlement(&mut self) -> Result<bool, SettleMatchExternalTaskError> {
-        todo!()
+        let valid_reblind = &self.internal_order_validity_bundle.reblind_proof.statement;
+        let nullifier = valid_reblind.original_shares_nullifier;
+        let res =
+            self.arbitrum_client.await_nullifier_spent(nullifier, ATOMIC_SETTLEMENT_TIMEOUT).await;
+
+        let did_settle = res.is_ok();
+        if !did_settle {
+            warn!("atomic match settlement not observed on-chain");
+        }
+        Ok(did_settle)
     }
 
     /// Refresh the wallet after the match settlement is witnessed on-chain
+    ///
+    /// Instead of replicating the logic of the refresh task we simply enqueue a
+    /// refresh task for the wallet.
+    ///
+    /// The wallet should have no other tasks enqueued by virtue of it being
+    /// considered for atomic matches, but if it does they will fail through
+    /// and be discarded
```

Comment:
this part of the doc comment is confusing to me, why is it the case that no other tasks should be enqueued? iirc there is no task queue clearing ahead of an external match settlement task (or any immediate task)

Reply:
because an order will not be selected as a match candidate if its task queue is not empty